pkg/tools/etcd_helper.go

```diff
@@ -469,7 +471,16 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 
 // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
 // See the comment for GuaranteedUpdate for more detail.
-type EtcdUpdateFunc func(input runtime.Object) (output runtime.Object, ttl uint64, err error)
+type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node, res *etcd.Response) (output runtime.Object, ttl *uint64, err error)
```

Comment:
That said I have no issue with changing the second and third are to a struct with a single member TTL for now.  Keeps us isolated